### PR TITLE
BaSP-TG: Added controller, model, validations and routes for Employees.

### PR DESCRIPTION
**Pull request by javierperezraffo**

### CODE REVIEW

File: src/models/Employees.js [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
 
-const { Schema } = mongoose;
\ No newline at end of file
+const { Schema } = mongoose;
+
+const employeeSchema = new Schema({
+  name: { type: String, required: true },
+  lastName: { type: String, required: true },
+  phone: { type: Number, required: true },
+  email: { type: String, required: true },
```
Comment (HoracioDuarteOK) [inline src/models/Employees.js:9]: You can add email:{ unique: true..

File: src/controllers/employees.js [hunk 2]
```diff
@@ -1,99 +1,107 @@
...
+  Employees.create(newParams, (error, employee) => {
+    if (error) {
+      res.status(400).json({
+        message: error.toString(),
+        data: undefined,
+        error: true,
+      });
     } else {
-      res.status(201).send('Employee Created');
+      res.status(200).json({
```
Comment (HoracioDuarteOK) [inline src/controllers/employees.js:51]: status 201

**Review by HoracioDuarteOK — APPROVED** [review]

File: postman_collection.json [hunk 3]
```diff
@@ -12,7 +11,113 @@
 		},
 		{
 			"name": "employee",
-			"item": []
+			"item": [
+				{
+					"name": "/",
```
Comment (Ignacio-Sormani) [inline postman_collection.json:16]: Add a name for each endpoint: GET ALL, GET BY ID, ADD, UPDATE and DELETE

File: postman_collection.json [hunk 4]
```diff
@@ -12,7 +11,113 @@
...
+					},
+					"response": []
+				},
+				{
+					"name": "/:id",
+					"request": {
+						"method": "GET",
+						"header": [],
+						"url": {
+							"raw": "localhost:3000/employees/634c94d68b20f4f1288dd4ef",
```
Comment (Ignacio-Sormani) [inline postman_collection.json:40]: It should be "localhost:3000/employees/:id". Add the id as a param in postman

Comment (Ignacio-Sormani) [reply postman_collection.json:40]: Do the same por put and delete methods

File: postman_collection.json [hunk 5]
```diff
@@ -1,9 +1,8 @@
 {
```
Comment (Ignacio-Sormani) [inline postman_collection.json:1]: Add a body example for post and put methods

File: src/controllers/employees.js [hunk 6]
```diff
@@ -1,99 +1,107 @@
...
+import Employees from '../models/Employees';
 
-const router = express.Router();
-
-router.get('/getAll', (req, res) => {
-  res.status(200).send(employees);
-});
-
-router.get('/getById/:id', (req, res) => {
+const getOne = (req, res) => {
```
Comment (Ignacio-Sormani) [inline src/controllers/employees.js:3]: All these functions are asynchronous. So you should use async with try/catch. Please take a look at you teammates PRs to find out how you should implement this

File: src/controllers/employees.js [hunk 7]
```diff
@@ -1,99 +1,107 @@
...
   const empId = req.params.id;
```
Comment (Ignacio-Sormani) [inline src/controllers/employees.js:4]: There is no need to store the employeeId in a const. You can just use it in line 5

File: src/controllers/employees.js [hunk 8]
```diff
@@ -1,99 +1,107 @@
...
+        error: false,
+      });
     }
   });
-});
+};
 
-router.put('/editEmployee/:id', (req, res) => {
+const edit = (req, res) => {
   const employeeId = req.params.id;
```
Comment (Ignacio-Sormani) [inline src/controllers/employees.js:61]: There is no need to store the employeeId

**Review by Ignacio-Sormani — CHANGES_REQUESTED** [review]

**Review by Ignacio-Sormani — APPROVED** [review]

**Pull request merged**